app.rb

```diff
@@ -0,0 +1,72 @@
+require 'sinatra'
+require 'sinatra/activerecord'
+require 'securerandom'
+
+enable :logging
+set :database, {adapter: "sqlite3", database: "foo.sqlite3"}
+set :bind, '127.0.0.1'
+set :server, 'thin'
+set :logger, Logger.new(STDOUT)
+
+class Scan < ActiveRecord::Base
+  validates_presence_of :target
+  validates_presence_of :port
+  validates_presence_of :state
+end
+
+class App < Sinatra::Base
+  before do
+    content_type :json
+  end
+
+  get '/queue' do
+    Scan.new do |s|
+      s.scan_id = SecureRandom.uuid
+      s.target = params["target"]
+      s.port = 22
+      s.state = "QUEUED"
+      s.save
```

Comment:
This is really so much cleaner and easier to read that the buried database abstraction layer.